collect_app/src/main/java/org/odk/collect/android/activities/FormChooserList.java

```diff
@@ -217,20 +214,27 @@ public void onClick(DialogInterface dialog, int i) {
         alertDialog.show();
     }
 
+    @NonNull
     @Override
     public Loader<Cursor> onCreateLoader(int id, Bundle args) {
-        progressBar.setVisibility(View.VISIBLE);
+        setToolbarProgress(true);
         return new FormsDao().getFormsCursorLoader(getFilterText(), getSortingOrder());
     }
 
     @Override
-    public void onLoadFinished(Loader<Cursor> loader, Cursor cursor) {
-        progressBar.setVisibility(View.GONE);
+    public void onLoadFinished(@NonNull Loader<Cursor> loader, Cursor cursor) {
+        updateProgressBar();
         listAdapter.changeCursor(cursor);
     }
 
     @Override
-    public void onLoaderReset(Loader loader) {
+    public void onLoaderReset(@NonNull Loader loader) {
         listAdapter.swapCursor(null);
     }
+
+    private void updateProgressBar() {
```

Comment:
A more descriptive name? `hideProgressBarWhenDone`?

Reply:
Thanks. Will surely update it